private/bufpkg/bufpluginexec/bufpluginexec.go

```diff
@@ -126,34 +130,42 @@ func GenerateWithProtocPath(protocPath string) GenerateOption {
 func NewHandler(
 	storageosProvider storageos.Provider,
 	runner command.Runner,
+	wasmPluginExecutor bufwasm.PluginExecutor,
 	pluginName string,
 	options ...HandlerOption,
 ) (appproto.Handler, error) {
 	handlerOptions := newHandlerOptions()
 	for _, option := range options {
 		option(handlerOptions)
 	}
+
+	// Initialize WASM plugin handler.
+	if looksLikeWASM(pluginName) {
+		return newWasmHandler(wasmPluginExecutor, pluginName), nil
+	}
+
+	// Initialize binary plugin handler when path is specified with optional args. Return
+	// on error as something is wrong with the supplied pluginPath option.
 	if len(handlerOptions.pluginPath) > 0 {
-		pluginPath, err := unsafeLookPath(handlerOptions.pluginPath[0])
-		if err != nil {
-			return nil, err
-		}
-		return newBinaryHandler(runner, pluginPath, handlerOptions.pluginPath[1:]), nil
+		return NewBinaryHandler(runner, handlerOptions.pluginPath[0], handlerOptions.pluginPath[1:])
```

Comment:
I did that because I noticed the exported version version does the error handling that I removed and calls `newProtocProxyHandler` which is not needed in the other two cases.